plugins/unpack/unpack.php

```diff
@@ -433,20 +538,27 @@ public function startTask( $hash, $outPath, $mode, $fileno )
 				{
 				        $rarPresent = false;
 				        $zipPresent = false;
+					$un7zPresent = false;
 					foreach($req->val as $no=>$name)
 					{
 						if(USE_UNRAR && (preg_match("'.*\.(rar|r\d\d|\d\d\d)$'si", $name)==1))
 							$rarPresent = true;
 						else
 						if(USE_UNZIP && (preg_match("'.*\.zip$'si", $name)==1))
 							$zipPresent = true;
+						else
+						if(USE_7Z && (preg_match("'.*\.7z$'si", $name)==1))
+							$un7zPresent = true;
 					}
-					$mode = ($rarPresent && $zipPresent) ? 'all' : ($rarPresent ? 'rar' : ($zipPresent ? 'zip' : null));
+					$mode = ($rarPresent && $zipPresent && $un7zPrenset) ? 'all' : ($rarPresent ? 'rar' : ($zipPresent ? 'zip' : ($un7zPresent ? '7z' : null)));
```

Comment:
What, if we will have 2 types of archives available in torrent? For example, only zip and rar?
As I may see, in this case only rar files will be unarchived. This is not a correct behavior, IMHO.

Reply:
I appreciate you taking the time to look further into this. As I said I just did a quick test and was able to unzip a 7z archive with it by changing the first issue you pointed out. I haven't looked that far into it other than that.

Reply:
Sorry, but 'quick tests' are not required.
If you make pull request, then this request must contain correct code. And you should be ready to support this code.